tests/mlmodel_sklearn/test_cluster_models.py

```diff
@@ -0,0 +1,186 @@
+    validate_transaction_metrics,
+)
+
+from newrelic.api.background_task import background_task
+from newrelic.common.package_version_utils import get_package_version
+from newrelic.packages import six
+
+SKLEARN_VERSION = tuple(map(int, get_package_version("sklearn").split(".")))
+
+
+@pytest.mark.parametrize(
+    "cluster_model_name",
+    [
+        "AffinityPropagation",
+        "AgglomerativeClustering",
+        "Birch",
+        "DBSCAN",
+        "FeatureAgglomeration",
+        "KMeans",
+        "MeanShift",
+        "MiniBatchKMeans",
+        "SpectralBiclustering",
+        "SpectralCoclustering",
+        "SpectralClustering",
+    ],
+)
+def test_below_v1_1_model_methods_wrapped_in_function_trace(cluster_model_name, run_cluster_model):
+    expected_scoped_metrics = {
+        "AffinityPropagation": [
+            ("Function/MLModel/Sklearn/Named/AffinityPropagation.fit", 2),
+            ("Function/MLModel/Sklearn/Named/AffinityPropagation.predict", 1),
+            ("Function/MLModel/Sklearn/Named/AffinityPropagation.fit_predict", 1),
+        ],
+        "AgglomerativeClustering": [
+            ("Function/MLModel/Sklearn/Named/AgglomerativeClustering.fit", 2),
+            ("Function/MLModel/Sklearn/Named/AgglomerativeClustering.fit_predict", 1),
+        ],
+        "Birch": [
+            ("Function/MLModel/Sklearn/Named/Birch.fit", 2),
+            (
+                "Function/MLModel/Sklearn/Named/Birch.predict",
+                1 if SKLEARN_VERSION >= (1, 0, 0) else 3,
+            ),
+            ("Function/MLModel/Sklearn/Named/Birch.fit_predict", 1),
+            ("Function/MLModel/Sklearn/Named/Birch.transform", 1),
+        ],
+        "DBSCAN": [
+            ("Function/MLModel/Sklearn/Named/DBSCAN.fit", 2),
+            ("Function/MLModel/Sklearn/Named/DBSCAN.fit_predict", 1),
+        ],
+        "FeatureAgglomeration": [
+            ("Function/MLModel/Sklearn/Named/FeatureAgglomeration.fit", 1),
+            ("Function/MLModel/Sklearn/Named/FeatureAgglomeration.transform", 1),
+        ],
+        "KMeans": [
+            ("Function/MLModel/Sklearn/Named/KMeans.fit", 2),
+            ("Function/MLModel/Sklearn/Named/KMeans.predict", 1),
+            ("Function/MLModel/Sklearn/Named/KMeans.fit_predict", 1),
+            ("Function/MLModel/Sklearn/Named/KMeans.transform", 1),
+        ],
+        "MeanShift": [
+            ("Function/MLModel/Sklearn/Named/MeanShift.fit", 2),
+            ("Function/MLModel/Sklearn/Named/MeanShift.predict", 1),
+            ("Function/MLModel/Sklearn/Named/MeanShift.fit_predict", 1),
+        ],
+        "MiniBatchKMeans": [
+            ("Function/MLModel/Sklearn/Named/MiniBatchKMeans.fit", 2),
+            ("Function/MLModel/Sklearn/Named/MiniBatchKMeans.predict", 1),
+            ("Function/MLModel/Sklearn/Named/MiniBatchKMeans.fit_predict", 1),
+        ],
+        "SpectralBiclustering": [
+            ("Function/MLModel/Sklearn/Named/SpectralBiclustering.fit", 1),
```

Comment:
If these only have a fit method they probably aren't a true model and we shouldn't be instrumenting them as a customer wouldn't use these directly. Let's double check with @ykriger-newrelic though.

Reply:
That's right, we don't have to instrument this model.

Reply:
Since we are instrumenting models that have fit (and not necessarily fit and predict) we do need to keep these.